src/components/inputs/ColorSelect/ColorSelect.test.tsx

```diff
@@ -0,0 +1,225 @@
+import React, { ReactElement } from 'react';
+import '@testing-library/jest-dom';
+import {
+  fireEvent,
+  render,
+  RenderOptions,
+  screen,
+} from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { ColorSelect } from './ColorSelect';
+import { ThemeProvider } from 'styled-components';
+import { themes } from '../../../themes';
+
+const Provider = ({ children }) => {
+  return (
+    <ThemeProvider theme={themes['light']}>{children}</ThemeProvider>
+  );
+};
+
+const renderWithThemeProvider = (
+  ui: ReactElement,
+  options?: Omit<RenderOptions, 'wrapper'>
+) => render(ui, { wrapper: Provider, ...options });
+
+const TEST_COLOR = '#CCCCCC';
+
+describe('ColorSelect', () => {
+  it('Renders ColorSelect input', () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByLabelText('color');
+    expect(input).toBeInTheDocument();
+  });
+
+  it('Renders ColorSelect picker using input', async () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+    const picker = screen.getByLabelText('color picker');
+    expect(picker).toBeInTheDocument();
+  });
+
+  it('Renders ColorSelect picker using children', async () => {
+    renderWithThemeProvider(
+      <ColorSelect>
+        <button data-testid="trigger" />
+      </ColorSelect>
+    );
+    const trigger = screen.getByTestId('trigger');
+    await userEvent.click(trigger);
+    const picker = screen.getByLabelText('color picker');
+    expect(picker).toBeInTheDocument();
+  });
+
+  it('Set ColorSelect color using value prop', async () => {
+    renderWithThemeProvider(<ColorSelect value={TEST_COLOR} />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    const selectedColor = screen.getByLabelText('color preview');
+    const selectedColorValue = selectedColor.getAttribute('color');
+    expect(selectedColorValue).toBe(TEST_COLOR);
+  });
+
+  it('Change ColorSelect color using input', async () => {
+    renderWithThemeProvider(<ColorSelect />);
+    const input = screen.getByLabelText('color');
+    await userEvent.click(input);
+
+    fireEvent.change(input, { target: { value: TEST_COLOR } });
```

Comment:
I'm not sure exactly why but `userEvent.type` / `userEvent.keyboard` are not working here.
From my search most issues are because changing testing-library from 13 to 14 that required await, this is not the case here.

When I look at how `userEvent.type` is [implemented](https://github.com/testing-library/user-event/blob/5feaa942f46bb37d96c2f2fbeb4b33e8beff75ad/src/type.js#L77) I see that there are checks for contenteditable and for [readonly](https://github.com/testing-library/user-event/blob/5feaa942f46bb37d96c2f2fbeb4b33e8beff75ad/src/type.js#L257).
I don't want to change current Input implementation as part of this PR, I think it should be addressed separately as part of the input tests because we'll probably have the same problem in every component using input.

Reply:
Ahh, yeah. that makes sense, thanks for the explanation. Best to address this in a separate PR. 😞